tmt/steps/provision/bootc.py

```diff
@@ -0,0 +1,232 @@
+            ln -s ../cloud-init.target /usr/lib/systemd/system/default.target.wants && \
+            rm /usr/local -rf && ln -sr /var/usrlocal /usr/local && mkdir -p /var/usrlocal/bin && \
+            dnf clean all
+        ''')
+        containerfile = containerfile_template.substitute({"base_image": base_image})
+        with open(f'{self.workdir}/Containerfile', 'w') as file:
+            file.write(containerfile)
+
+        image_tag = f'localhost/tmtmodified-{self._id}'
+        tmt.utils.Command(
+            "podman", "build", f'{self.workdir}',
+            "-f", f'{self.workdir}/Containerfile',
+            "-t", image_tag
+            ).run(cwd=self.workdir, stream_output=True, logger=self._logger)
+
+        return image_tag
+
+    # build the "base" or user supplied container image
+    def _build_base_image(self, containerfile: str, workdir: str) -> str:
+        image_tag = f'localhost/tmtbase-{self._id}'
+        self._logger.debug("Building container image")
+        tmt.utils.Command(
+            "podman", "build", workdir,
+            "-f", containerfile,
+            "-t", image_tag
+            ).run(cwd=self.workdir, stream_output=True, logger=self._logger)
+        return image_tag
+
+    # build the bootc disk
+    def _build_bootc_disk(self, containerimage: str, image_builder: str) -> None:
+        self._logger.debug("Building bootc disk image")
+        tmt.utils.Command(
+            "podman", "run", "--rm", "--privileged",
+            "-v", "/var/lib/containers/storage:/var/lib/containers/storage",
+            "--security-opt", "label=type:unconfined_t",
+            "-v", f"{self.workdir}:/output",
+            image_builder, "build",
+            "--type", "qcow2",
+            "--local", containerimage
+            ).run(cwd=self.workdir, stream_output=True, logger=self._logger)
+
+    def go(self, *, logger: Optional[tmt.log.Logger] = None) -> None:
+        """ Provision the bootc instance """
+        super().go(logger=logger)
+
+        data = BootcData.from_plugin(self)
+        data.set_image(f"file://{self.workdir}/qcow2/disk.qcow2")
+        data.show(verbose=self.verbosity_level, logger=self._logger)
+
+        containerimage = ""
+        if data.containerimage is not None:
+            containerimage = data.containerimage
+            if data.add_deps:
+                containerimage = self._build_derived_image(data.containerimage)
+            self._build_bootc_disk(containerimage, data.image_builder)
+        elif data.containerfile is not None:
+            containerimage = self._build_base_image(data.containerfile, data.containerfile_workdir)
+            if data.add_deps:
+                containerimage = self._build_derived_image(containerimage)
+            self._build_bootc_disk(containerimage, data.image_builder)
+        else:
+            self._logger.fail("Either containerfile or containerimage must be specified.")
+            raise SystemExit(1)
```

Comment:
This is definitely worth an exception, e.g. `raise tmt.utils.ProvisionError("Either containerfile or containerimage must be specified.")`. tmt has some basic exception rendering, please leave the reporting and exit strategy to tmt core, it will be fine :)